tfhe/src/integer/gpu/list_compression/server_keys.rs

```diff
@@ -0,0 +1,328 @@
+use crate::core_crypto::gpu::entities::lwe_packing_keyswitch_key::CudaLwePackingKeyswitchKey;
+use crate::core_crypto::gpu::glwe_ciphertext_list::CudaGlweCiphertextList;
+use crate::core_crypto::gpu::lwe_bootstrap_key::CudaLweBootstrapKey;
+use crate::core_crypto::gpu::lwe_ciphertext_list::CudaLweCiphertextList;
+use crate::core_crypto::gpu::vec::CudaVec;
+use crate::core_crypto::gpu::CudaStreams;
+use crate::core_crypto::prelude::{
+    allocate_and_generate_new_lwe_packing_keyswitch_key,
+    par_allocate_and_generate_new_lwe_bootstrap_key, CiphertextModulusLog, GlweCiphertextCount,
+    LweBootstrapKeyOwned, LweCiphertextCount, LwePackingKeyswitchKey,
+};
+use crate::integer::gpu::ciphertext::info::{CudaBlockInfo, CudaRadixCiphertextInfo};
+use crate::integer::gpu::ciphertext::CudaRadixCiphertext;
+use crate::integer::gpu::server_key::CudaBootstrappingKey;
+use crate::integer::gpu::{
+    compression_compress_integer_radix_async, compression_decompress_integer_radix_async,
+    cuda_memcpy_async_gpu_to_gpu,
+};
+use crate::integer::parameters::{GlweDimension, PolynomialSize};
+use crate::shortint::client_key::ClientKey;
+use crate::shortint::engine::ShortintEngine;
+use crate::shortint::list_compression::{CompressionKey, CompressionPrivateKeys};
+use crate::shortint::{ClassicPBSParameters, EncryptionKeyChoice, PBSParameters};
+use itertools::Itertools;
+
+#[derive(Debug)]
+pub struct CudaCompressionKey {
+    pub packing_key_switching_key: CudaLwePackingKeyswitchKey<u64>,
+    pub h_packing_key_switching_key: LwePackingKeyswitchKey<Vec<u64>>,
```

Comment:
Why keep the CPU key here? I think we want to have only the Cuda key here.

Reply:
Another debug artifact. Sorry.